index.js

```diff
@@ -1,23 +1,48 @@
 const stylelint = require('stylelint');
 
-const ruleName = '@apostrophecms/require-nested-after-include';
+const ruleName = '@apostrophecms/stylelint-no-mixed-decls';
 const messages = stylelint.utils.ruleMessages(ruleName, {
-  expected: 'Expected "& { ... }" block after "@include" at-rule if declarations are present. See https://sass-lang.com/documentation/breaking-changes/mixed-decls/'
+  mixed: 'Cannot mix declarations and nested rules/at-rules. Group them together or wrap declarations in a nested "& { }" block. See https://sass-lang.com/documentation/breaking-changes/mixed-decls/'
 });
 
-module.exports = stylelint.createPlugin(ruleName, () => {
+module.exports = stylelint.createPlugin(ruleName, (primary, secondaryOptions = {}) => {
+  const safeMixins = secondaryOptions.safeMixins || [];
   return (root, result) => {
-    root.walkRules((rule) => {
-      rule.walkAtRules('include', (atRule) => {
-        const nextNode = atRule.next();
+    root.walkRules(rule => {
+      let seenDeclaration = false;
+      let seenNested = false;
 
-        if (nextNode?.type === 'decl') {
-          stylelint.utils.report({
-            message: messages.expected,
-            node: atRule,
-            result,
-            ruleName
-          });
+      rule.each(node => {
+        if (node.type === 'decl') {
+          if (seenNested) {
+            stylelint.utils.report({
+              message: messages.mixed,
+              node,
+              result,
+              ruleName
+            });
+          }
+          seenDeclaration = true;
+        }
+
+        if (node.type === 'rule' && node.selector.startsWith('&')) {
+          seenNested = true;
+        }
+
+        if (node.type === 'atrule' && node.name === 'include') {
+          const isSafeMixin = safeMixins.includes(node.params.trim());
+
+          if (!isSafeMixin) {
+            if (seenDeclaration) {
```

Comment:
`if (!isSafeMixin & seenDeclaration) { ... }` is more readable and better here, because you don't have any `else` statements for both expressions.

Reply:
there's the `seenNested = true;` line 44 that's in the `if (!isSafeMixin)` block

Reply:
OK, it was hard to see in the diff view.